claripy/ast/base.py

```diff
@@ -112,8 +112,36 @@ def __new__(cls, op, args, add_variables=None, hash=None, **kwargs): #pylint:dis
 
         a_args = args if type(args) is tuple else tuple(args)
 
-        # initialize the following properties: symbolic, variables and errored
+        # fast path: do not hash or cache BVV
```

Comment:
There are a lot of places in angr where we use `is` to reason about equality of BVVs, especially (but not only) in the simplifiers. This will break that. If we want this route, we should implement something like `ast.is_same` to replace using `is`, though that might slow things down significantly...

Reply:
In the next angr major version we should have == check structural equality (with fast paths of course) and add a .equals for the comparison operation.

Reply:
That makes me nervous... Having an `==` that can't return a BoolS will inject the sort of confusion that happens over `is_true` and `is_false`, but to everything, and needing to call `.equals` a) gives me Java PTSD and b) means `a.equals(b)` could AttributeError out when `a` is an int, whereas `a == b` currently won't.